comp/core/autodiscovery/listeners/snmp.go

```diff
@@ -408,6 +406,11 @@ func (l *SNMPListener) checkDevices() {
 	discoveryTicker := time.NewTicker(time.Duration(l.config.DiscoveryInterval) * time.Second)
 	defer discoveryTicker.Stop()
 	for {
+		// Reset the device deduper counters at the start of each discovery interval
+		if l.deviceDeduper != nil {
```

Comment:
Do we need this check? In the rest of this file we expect deviceDeduper to be non-nil.
Is there a chance to call checkDevices with a non-initialized deviceDeduper? Can we prevent this if that's an option?